core/src/main/java/com/linecorp/armeria/client/retry/BackoffSpec.java

```diff
@@ -0,0 +1,300 @@
+        jitterConfigured = true;
+
+        final List<String> values = VALUE_SPLITTER.splitToList(jitterValues);
+        checkValueSize(key, values.size(), 3, jitterValues);
+
+        if (values.size() == 1) {
+            double jitterRate = parseDouble(key, values.get(0), ORDINAL[0]);
+            checkDoubleBetween(key, jitterRate, 0.0, 1.0, ORDINAL[0]);
+            minJitterRate = jitterRate * -1;
+            maxJitterRate = jitterRate;
+        } else {
+            minJitterRate = parseDouble(key, values.get(0), ORDINAL[0]);
+            checkDoubleBetween(key, minJitterRate, -1.0, 1.0, ORDINAL[0]);
+            maxJitterRate = parseDouble(key, values.get(1), ORDINAL[1]);
+            checkDoubleBetween(key, maxJitterRate, -1.0, 1.0, ORDINAL[1]);
+            if (minJitterRate > maxJitterRate) {
+                double temp = minJitterRate;
+                minJitterRate = maxJitterRate;
+                maxJitterRate = temp;
+            }
+        }
+    }
+
+    private void checkDoubleBetween(String key, double value, double min, double max, String ordinal) {
+        checkArgument(min <= value && value <= max,
+                      "%s parameter for %s must be >= %s and <= %s. input: %s",
+                      ordinal, key, min, max, value);
+    }
+
+    private void maxAttempts(String key, String value) {
+        checkArgument(!maxAttemptsConfigured, "maxAttempts parameters is already set. maxAttempts: %s",
+                      maxAttempts);
+        maxAttemptsConfigured = true;
+        maxAttempts = parseInt(key, value);
+        checkNegative(key, maxAttempts, ORDINAL[0]);
+    }
+
+    private static long parseLong(String key, String value, String ordinal) {
+        try {
+            return Long.parseLong(value);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(String.format(
+                    "%s parameter for %s was set to %s, must be a long", ordinal, key, value), e);
+        }
+    }
+
+    private static double parseDouble(String key, String value, String ordinal) {
+        try {
+            return Double.parseDouble(value);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(String.format(
+                    "%s parameter for %s was set to %s, must be a double", ordinal, key, value), e);
+        }
+    }
+
+    private static int parseInt(String key, String value) {
+        try {
+            return Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(
+                    String.format("%s was set to %s, must be an integer", key, value), e);
+        }
+    }
+
+    /**
+     * Returns a newly-created {@link Backoff} based on the properties of this spec.
+     */
+    Backoff build() {
+        Backoff backoff =
+                !fixedConfigured && !randomConfigured ?
+                Backoff.exponential(initialDelayMillis, maxDelayMillis, multiplier)
```

Comment:
The tertiary is long so I'm not even sure, but I think this can be more readable as

```java
if (fixedConfigured)

else if (random)

else
```